### PR TITLE
Basic CTAS Implementation

### PR DESCRIPTION
**Pull request by sundarshankar89**
Basic CTAS Implementation for Snowflake

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala [hunk 1]
```diff
@@ -100,8 +100,8 @@ class LogicalPlanGenerator(
         }
 
         s"${leadingComment}CREATE TABLE ${ct.table_name} (${columns}${tableConstraintStrWithComma})"
-
-      case ctas: ir.CreateTableAsSelect => s"CREATE TABLE ${ctas.table_name} AS (${generate(ctx, ctas.query)})"
+      // Removed Unnecessary Braces
```
Comment (jimidle) [inline core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala:103]: I think we can discard this comment - a minor nit ;)

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/ddl.scala [hunk 2]
```diff
@@ -155,6 +155,7 @@ case class CreateTable(
     extends Catalog {}
 
 // As per Spark v2Commands
+// TODO - This IR needs to be extended to Support Table Properties, Partitioning, Clustering etc.
```
Comment (jimidle) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/ddl.scala:158]: You can follow the way that TSQL does it - wrapping this base IR with additions

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeDDLBuilder.scala [hunk 3]
```diff
@@ -92,6 +93,23 @@ class SnowflakeDDLBuilder
     ir.CreateTableCommand(tableName, columns)
   }
 
+  override def visitCreateTableAsSelect(ctx: CreateTableAsSelectContext): ir.Catalog = {
+    val tableName = ctx.objectName().getText
+    val selectStatement = ctx.queryStatement().accept(relationBuilder)
+    // Currently TableType is not used in the IR and Databricks doesn't support Temporary Tables
+    // val tableType = ctx.tableType().getText
```
Comment (jimidle) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeDDLBuilder.scala:100]: Generally leaving in commented code is not necessary

**Review by jimidle — CHANGES_REQUESTED** [review]
Just a few minor cleanups but lgtm other than that

File: core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala (hunk 1 above)
Comment (sundarshankar89) [reply core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala:103]: done

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**